File: backend/models/articles.py
from bson import ObjectId

def article_schema(data):
    return {
        "_id": str(data.get("_id", ObjectId())),
        "titulo": data.get("titulo", ""),
        "contenido_markdown": data.get("contenido_markdown", ""),
        "imagen_url": data.get("imagen_url", ""),
        "tags": data.get("tags", []),
        "autor_id": data.get("autor_id", ""),
        "fecha_creacion": data.get("fecha_creacion"),
        "descripcion": data.get("descripcion", ""),
    }

File: frontend/src/api/articles.ts
import api from "./axios";

export interface Article {
    _id: string;
    titulo: string;
    contenido_markdown: string;
    imagen_url: string;
    autor_id: string;
    fecha_creacion: string;
    tags: string[];
}

export const getArticles = async (): Promise<Article[]> => {
    const res = await api.get("/articles/");
    return res.data;
};
